internal/cluster/failover_service.go

```diff
@@ -24,63 +24,57 @@ import (
 	"github.com/hazelcast/hazelcast-go-client/internal/security"
 )
 
-// Responsible for cluster failover state and attempts management.
+// FailoverService is responsible for cluster failover state and attempts management.
 type FailoverService struct {
-	isClientRunningFn func() bool
+	clientRunningFn   func() bool
 	candidateClusters []CandidateCluster
 	maxTryCount       int
 	index             uint64
 }
 
 type CandidateCluster struct {
-	AddressProvider   AddressProvider
-	AddressTranslator AddressTranslator
-	Credentials       security.Credentials
-	ClusterName       string
+	AddressProvider    AddressProvider
+	AddressTranslator  AddressTranslator
+	Credentials        security.Credentials
+	ConnectionStrategy *pubcluster.ConnectionStrategyConfig
+	ClusterName        string
 }
 
-func NewFailoverService(
-	logger ilogger.Logger,
-	maxTryCount int,
-	rootConfig pubcluster.Config,
-	failoverConfigs []pubcluster.Config,
-	addrProviderTranslatorFn func(*pubcluster.Config, ilogger.Logger) (AddressProvider, AddressTranslator),
-	isClientRunningFn func() bool) *FailoverService {
+type addrFun func(*pubcluster.Config, ilogger.Logger) (AddressProvider, AddressTranslator)
 
-	candidateClusters := []CandidateCluster{}
-	configs := []pubcluster.Config{rootConfig}
-	configs = append(configs, failoverConfigs...)
+func NewFailoverService(logger ilogger.Logger, maxTries int, rootConfig pubcluster.Config, foConfigs []pubcluster.Config, addrFn addrFun, clientRunningFn func() bool) *FailoverService {
+	candidates := []CandidateCluster{}
+	configs := []pubcluster.Config{}
+	if len(foConfigs) > 0 {
+		configs = foConfigs
+	} else {
+		configs = append(configs, rootConfig)
+	}
 	for _, c := range configs {
-		ctx := CandidateCluster{
-			ClusterName: c.Name,
-			Credentials: makeCredentials(&c.Security),
+		// copy c to a local variable in order to be able to take its address below
+		cv := c
+		cc := CandidateCluster{
+			ClusterName:        c.Name,
+			Credentials:        makeCredentials(&c.Security),
+			ConnectionStrategy: &cv.ConnectionStrategy,
 		}
-		ctx.AddressProvider, ctx.AddressTranslator = addrProviderTranslatorFn(&c, logger)
-		candidateClusters = append(candidateClusters, ctx)
+		cc.AddressProvider, cc.AddressTranslator = addrFn(&c, logger)
+		candidates = append(candidates, cc)
 	}
 
 	return &FailoverService{
-		isClientRunningFn: isClientRunningFn,
-		maxTryCount:       maxTryCount,
-		candidateClusters: candidateClusters,
+		clientRunningFn:   clientRunningFn,
```

Comment:
It looks like this field is not used anymore.

Reply:
Removed in: https://github.com/hazelcast/hazelcast-go-client/pull/615/commits/12c18628c7cb62180f904ad1f6e6c98a8a96a77e